src/components/PopupWithForm.js

```diff
@@ -4,27 +4,33 @@ class PopupWithForm extends Popup {
   constructor(popupSelector, handleFormSubmit) {
     super({ popupSelector });
     this._popupForm = this._modalEl.querySelector(".modal__form");
+    this._submitButton = this._modalEl.querySelector(".modal__save-button");
     this._handleFormSubmit = handleFormSubmit;
     this._list = [...this._popupForm.querySelectorAll(".modal__input")];
   }
 
+  open() {
+    this._submitButton.textContent = "Save";
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.